src/main/java/org/opensearch/sdk/ExtensionApi.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.sdk;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * This class encapsulates the API of an Extension.
+ */
+public class ExtensionApi {
```

Comment:
@saratvemulapalli Do we have a style that would prefer `RESTApi` or `RESTAPI` or `RestApi`?

Reply:
I went with `RestApi` because while a standalone `RestAPI` might work it's also used in longer classes like `RestApiResponseHandler` and I strongly prefer the lowercase there.   But @joshpalis has used `NamedWriteableRegistryAPI` so we should align on a style before we create any more APIs.

Reply:
Within the OpenSearch codebase, the moniker is `Rest`, as in `RestController.java`

Names / suffixes that I do not recommend using, `Framework`, `Utility`, `Helper`, and `API` they tend not to impart any additional information about the what or why of the class.  

As I see the utility of this class, it seems like its a registry of extension specific actions/operations.  Maybe something along the lines of `RegisteredActions` or `AvailableOperations` - What do you think?

Reply:
Great observations, @peternied.  I agree that for many of these we can get rid of the API and I'll work on that later today.  My thoughts (mostly notes to myself for later):
 - I think I'll leave the `ExtensionRestApi` class named as-is for now, as it's really a placeholder class for reading a YAML file, both of which will eventually go away when we eventually implement an OpenAPI-compatible spec file.  And given the strong intended tie to "OpenAPI" I'd like to keep the API (or Api) in the name for now.
 - The remainder of the operations are going to be "actions" that need to be "handled", e.g, `public class RestDeleteAnomalyDetectorAction extends BaseRestHandler`.   I think either `FooExtensionRestAction` or `FooExtensionRestHandler` or the similar for ...Request and ...Response will be a better convention.

FYI @joshpalis I think you should consider removing API from the NamedWriteables code for similar reasons.

FYI @owaiskazi19 and @saratvemulapalli when you hopefully review this PR tomorrow.

Reply:
That makes a lot of sense. May be we can have a legend for conventions something like[1]. 
I agree it makes sense to have them as `ExtensionRestApi` 

[1] https://github.com/opensearch-project/opensearch-plugins/blob/main/CONVENTIONS.md